post.js

```diff
@@ -106,18 +106,31 @@ class DistributedPost extends HTMLElement {
     postContent.classList.add('post-content')
 
     // Determine content source based on structure of jsonLdData
-    let contentSource = jsonLdData.content || (jsonLdData.object && jsonLdData.object.content)
-    // Check if the content has a mention and update the href attribute
-    if (jsonLdData.tag && jsonLdData.tag.some(tag => tag.type === 'Mention')) {
-      console.log(jsonLdData.tag.type)
-      jsonLdData.tag.forEach(tag => {
-        if (tag.type === 'Mention') {
-          // Replace the href with the profile link
-          const regex = new RegExp(`href="${tag.href}"`, 'g')
-          contentSource = contentSource.replace(regex, `href="/profile.html?actor=${encodeURIComponent(tag.href)}"`)
+    const contentSource = jsonLdData.content || (jsonLdData.object && jsonLdData.object.content)
+
+    // Sanitize content and create a DOM from it
+    const sanitizedContent = DOMPurify.sanitize(contentSource)
+    const parser = new DOMParser()
+    const contentDOM = parser.parseFromString(sanitizedContent, 'text/html')
+
+    // Process all anchor elements
+    const anchors = contentDOM.querySelectorAll('a')
+    anchors.forEach(anchor => {
+      const href = anchor.getAttribute('href')
+      if (href) {
+        const fediverseMatch = href.match(/^https?:\/\/([^\/]+)\/@(\w+)$/)
```

Comment:
This is not sufficient. We need to use the DB to attempt to resolve the URL to an activity or actor by doing a request. I could do this bit if you'd like help.